src/test/java/hudson/plugins/xvnc/XvncTest.java

```diff
@@ -134,6 +147,118 @@ public void avoidNpeAfterDeserialiation() throws Exception {
         j.buildAndAssertSuccess(p);
     }
 
+    public void testXauthorityInWorkspace() throws Exception {
+        DumbSlave slave = j.createOnlineSlave();
+
+        FreeStyleProject job = j.jenkins.createProject(FreeStyleProject.class, "jobA");
+        job.setAssignedNode(slave);
+
+        runXvnc(job, true);
+
+        job.getBuildersList().add(new TestBuilder() {
+            @Override
+            public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
+                FilePath[] list = build.getWorkspace().list(".Xauthority-*");
+                assertEquals("There should be one Xauthority file in the workspace", 1, list.length);
+                assertEquals(build.getWorkspace().getRemote(), list[0].getParent().getRemote());
+                return true;
+            }
+        });
+        j.buildAndAssertSuccess(job);
+
+        FilePath[] list = job.getLastBuild().getWorkspace().list(".Xauthority-*");
+        assertEquals("The Xauthority file should be removed", 0, list.length);
+    }
+
+    @Test
+    public void testXauthorityInSlaveFsRoot() throws Exception {
+        final DumbSlave slave = j.createOnlineSlave();
+
+        FreeStyleProject job = j.jenkins.createProject(FreeStyleProject.class, "job A");
+        job.setAssignedNode(slave);
+
+        runXvnc(job, true);
+
+        job.getBuildersList().add(new TestBuilder() {
+            @Override
+            public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
+                FilePath[] list = slave.getRootPath().list(".Xauthority-*");
+                assertEquals("There should be one Xauthority file in the slave fs root", 1, list.length);
```

Comment:
the slave is created with its workspace in the temp directory IIRC when using JEnkinsRule - in which case if the temp dir contains a space this will fail.